### PR TITLE
all: Sanitize API Keys 

### LINKED ISSUE #6961: tracing-agent: API key sanitization
There are many places in Datadog Agent code base where API key is read from config structs. It is sanitized in [some cases](https://github.com/DataDog/datadog-agent/blob/570ac79334d1e0306485335ad8073481c9954159/pkg/trace/config/apply.go#L154), but [not others](https://github.com/DataDog/datadog-agent/blob/570ac79334d1e0306485335ad8073481c9954159/pkg/trace/config/apply.go#L125). That costed me quite some time debugging (see #6960). I wonder if this is intentional or should be fixed? It seems like centralizing the API key reads would be a good solution, but I'm reluctant to do the change because it'd touch a lot of files and is thus unlikely to get accepted.

### PR DESCRIPTION
**Pull request by nicoledanuwidjaja**
Standardizes API key usage by sanitizing all API key reads when config is set.

Fixes #6961.

### CODE REVIEW

**Review by knusbaum — APPROVED** [review]
LGTM.

**Review by mx-psi — APPROVED** [review]
LGTM, I launched pipeline 3802690 with the full test suite to double check and everything is passing

**Review by gbbr — APPROVED** [review]
LGTM for APM side

**Comment by remeh** [comment]
Thanks for the PR 👍 
I'm wondering if we shouldn't consider modifying our fork of Viper (used for the config) to have a special cases while reading some fields (i.e. here the `api_key`) and to automatically sanitize them.

**Review by remeh — APPROVED** [review]
Thanks for the PR 👍
I'm wondering if we shouldn't consider modifying our fork of Viper (used for the config) to have a special case while reading some fields (i.e. here the api_key) and to automatically sanitize them.

**Comment by gbbr** [comment]
Something's wrong with the pipeline here. @mx-psi any idea what happened? Everything's in "Expected" mode.

**Comment by mx-psi** [comment]
> Something's wrong with the pipeline here. @mx-psi any idea what happened? Everything's in "Expected" mode.

The CircleCI failed to run for some reason, it is a known issue and there is no known workaround other than opening a new PR. However, if you scroll down you can see the Gitlab and AppVeyor tests all have ran and are green. 

Gitlab tests should check for everything CircleCI tests check except linters (for Python and shell code, not applicable), team label (✅), release note (missing) and milestone (✅). So the only issue would be the release note.

**Comment by gbbr** [comment]
Thanks!

@nicoledanuwidjaja can you please add a release note? Instructions are here: https://github.com/DataDog/datadog-agent/blob/master/docs/dev/contributing.md#reno

**Comment by nicoledanuwidjaja** [comment]
You got it!

**Review by L3n41c — COMMENTED** [review]
Thanks Nicole for contributing this PR!
I’m just wondering if it would be simpler and less error prone to sanitise the API key only once when the configuration is parsed rather than each time this parameter is used.

It seems to already be done, in some cases, thanks to:
https://github.com/DataDog/datadog-agent/blob/7.25.1/pkg/config/config.go#L986-L989
```go
// SanitizeAPIKeyConfig strips newlines and other control characters from a given key.
func SanitizeAPIKeyConfig(config Config, key string) {
	config.Set(key, SanitizeAPIKey(config.GetString(key)))
}
```

which is called from:
* https://github.com/DataDog/datadog-agent/blob/7.25.1/pkg/config/config.go#L942
* https://github.com/DataDog/datadog-agent/blob/7.25.1/pkg/logs/config/config.go#L105

**Review by robertjli — APPROVED** [review]

File: pkg/process/config/yaml_config.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func (a *AgentConfig) LoadProcessYamlConfig(path string) error {
 	a.APIEndpoints[0].Endpoint = URL
 
 	if key := "api_key"; config.Datadog.IsSet(key) {
-		a.APIEndpoints[0].APIKey = config.Datadog.GetString(key)
+		a.APIEndpoints[0].APIKey = config.SanitizeAPIKey(config.Datadog.GetString(key))
```
Comment (robertjli) [inline pkg/process/config/yaml_config.go:241]: We have additional API keys on line 379 as well

**Comment by nicoledanuwidjaja** [comment]
@L3n41c 
> Thanks Nicole for contributing this PR!
> I’m just wondering if it would be simpler and less error prone to sanitise the API key only once when the configuration is parsed rather than each time this parameter is used.

Hello! Was initially going with that route but was unsure if this would cover all use cases.

**Review by blemale — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Review by hkaj — APPROVED** [review]

**Pull request merged**